pkg/provider/barbican/client.go

```diff
@@ -0,0 +1,236 @@
+	errFetchSAKSecret       = "could not fetch SecretAccessKey secret: %w"
+	errMissingSAK           = "missing SecretAccessKey"
+	errKeyNotFound          = "key '%s' does not exist"
+	errKeyAlreadyExists     = "key '%s' already exists"
+	errJSONSecretUnmarshal  = "unable to unmarshal secret: %w"
+	errInvalidStore         = "invalid store"
+	errInvalidStoreSpec     = "invalid store spec"
+	errInvalidStoreProv     = "invalid store provider"
+	errInvalidBarbicanProv  = "invalid barbican secrets manager provider"
+	errInvalidAuthSecretRef = "invalid auth secret ref: %w"
+)
+
+type Client struct {
+	client    *gophercloud.ServiceClient
+	kube      kclient.Client
+	store     *esv1beta1.BarbicanProvider
+	storeKind string
+
+	// namespace of the external secret
+	namespace string
+}
+
+func (c *Client) DeleteSecret(ctx context.Context, remoteRef esv1beta1.PushSecretRemoteRef) error {
+	var (
+		name   = remoteRef.GetRemoteKey()
+		uuid   = getUUIDFromRemoteRef(name)
+		secret *secrets.Secret
+		err    error
+	)
+
+	if uuid == "" {
+		secret, err = c.getByName(ctx, name)
+		if err != nil {
+			return err
+		}
+		uuid = extractIdFromRef(secret.SecretRef)
+	} else {
+		_, err = c.getByUUID(ctx, uuid)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = secrets.Delete(c.client, uuid).Err
+	metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanDeleteSecret, err)
+	return err
+}
+
+// GetAllSecrets syncs multiple secrets from barbican provider into a single Kubernetes Secret.
+func (c *Client) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	allPages, err := secrets.List(c.client, secrets.ListOpts{
+		Sort: "created:desc",
+	}).AllPages()
+	metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetAllSecrets, err)
+	if err != nil {
+		return nil, err
+	}
+
+	allSecrets, err := secrets.ExtractSecrets(allPages)
+
+	mapUUIDByUniqueNames := make(map[string]string, len(allSecrets))
+	for _, v := range allSecrets {
+		if _, ok := mapUUIDByUniqueNames[v.Name]; !ok {
+			mapUUIDByUniqueNames[v.Name] = extractIdFromRef(v.SecretRef)
+		}
+	}
+
+	result := make(map[string][]byte, len(mapUUIDByUniqueNames))
+
+	for name, id := range mapUUIDByUniqueNames {
+		payload, err := secrets.GetPayload(c.client, id, secrets.GetPayloadOpts{PayloadContentType: "*/*"}).Extract()
+		metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetSecretPayload, err)
+		if err != nil {
+			return nil, err
+		}
+
+		result[name] = payload
+	}
+
+	return result, nil
+}
+
+// GetSecret returns a single secret from the provider.
+func (c *Client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	var (
+		name   = ref.Key
+		uuid   = getUUIDFromRemoteRef(name)
+		secret *secrets.Secret
+		err    error
+	)
+
+	metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetSecret, err)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is checking the above declared error? I think this must have been moved here by accident.